### PR TITLE
fix: fix `remix.init`

### LINKED ISSUE #134: Fix JavaScript version
### Have you experienced this bug with the latest version of the template?

Yes

### Steps to Reproduce

Create a new project with this template and choose JavaScript.

### Expected Behavior

No errors when running `npm run validate`

### Actual Behavior

After @MichaelDeBoey's PR #127 it looks like there are still more fixes to be made.

I ran into a couple issues when running `npm run validate`: 

```
TypeError: tsconfigPaths is not a function
```

After fixing that, I got:

```

/Users/kentcdodds/Desktop/my-remix-app/app/routes/notes/$noteId.jsx
  6:1  warning  '~/models/note.server' import is duplicated  no-duplicate-imports

/Users/kentcdodds/Desktop/my-remix-app/cypress/e2e/smoke.cy.js
   3:1  error  'describe' is not defined   no-undef
   4:3  error  'afterEach' is not defined  no-undef
   5:5  error  'cy' is not defined         no-undef
   8:3  error  'it' is not defined         no-undef
  14:5  error  'cy' is not defined         no-undef
  16:5  error  'cy' is not defined         no-undef
  18:5  error  'cy' is not defined         no-undef
  20:5  error  'cy' is not defined         no-undef
  21:5  error  'cy' is not defined         no-undef
  22:5  error  'cy' is not defined         no-undef
  24:5  error  'cy' is not defined         no-undef
  25:5  error  'cy' is not defined         no-undef
  26:5  error  'cy' is not defined         no-undef
  29:3  error  'it' is not defined         no-undef
  35:5  error  'cy' is not defined         no-undef
  37:5  error  'cy' is not defined         no-undef
  39:5  error  'cy' is not defined         no-undef
  40:5  error  'cy' is not defined         no-undef
  42:5  error  'cy' is not defined         no-undef
  44:5  error  'cy' is not defined         no-undef
  45:5  error  'cy' is not defined         no-undef
  46:5  error  'cy' is not defined         no-undef
  48:5  error  'cy' is not defined         no-undef
  50:5  error  'cy' is not defined         no-undef

/Users/kentcdodds/Desktop/my-remix-app/cypress/support/commands.js
   6:3   error  'cy' is not defined                                                  no-undef
   7:3   error  'cy' is not defined                                                  no-undef
  13:5   error  'cy' is not defined                                                  no-undef
  15:10  error  'cy' is not defined                                                  no-undef
  22:5   error  'cy' is not defined                                                  no-undef
  29:3   error  'cy' is not defined                                                  no-undef
  33:3   error  'cy' is not defined                                                  no-undef
  37:3   error  'cy' is not defined                                                  no-undef
  46:3   error  'cy' is not defined                                                  no-undef
  47:3   error  'cy' is not defined                                                  no-undef
  50:1   error  'Cypress' is not defined                                             no-undef
  51:1   error  'Cypress' is not defined                                             no-undef
  52:1   error  'Cypress' is not defined                                             no-undef
  54:1   error  Definition for rule '@typescript-eslint/no-namespace' was not found  @typescript-eslint/no-namespace

/Users/kentcdodds/Desktop/my-remix-app/cypress/support/e2e.js
  4:1  error  'Cypress' is not defined  no-undef

✖ 40 problems (39 errors, 1 warning)
```

Running `npm run test:e2e:run` I got:

```
Your configFile is invalid: /Users/kentcdodds/Desktop/my-remix-app/cypress.config.js

It threw an error when required, check the stack trace below:

/Users/kentcdodds/Desktop/my-remix-app/cypress.config.js:3
export default defineConfig({
^^^^^^

SyntaxError: Unexpected token 'export'
    at Object.compileFunction (node:vm:353:18)
    at wrapSafe (node:internal/modules/cjs/loader:1040:15)
    at Module._compile (node:internal/modules/cjs/loader:1076:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1166:10)
    at Module.load (node:internal/modules/cjs/loader:988:32)
    at Module._load (node:internal/modules/cjs/loader:834:12)
    at Module.require (node:internal/modules/cjs/loader:1012:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at loadFile (/Users/kentcdodds/Library/Caches/Cypress/10.4.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/child/run_require_async_child.js:89:14)
    at EventEmitter.<anonymous> (/Users/kentcdodds/Library/Caches/Cypress/10.4.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/child/run_require_async_child.js:116:38)
    at EventEmitter.emit (node:events:537:28)
    at process.<anonymous> (/Users/kentcdodds/Library/Caches/Cypress/10.4.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/util.js:33:22)
    at process.emit (node:events:537:28)
    at emit (node:internal/child_process:937:14)
    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)

Error: Command failed with exit code 1: npx cypress run
    at makeError (/Users/kentcdodds/Desktop/my-remix-app/node_modules/start-server-and-test/node_modules/execa/lib/error.js:60:11)
    at handlePromise (/Users/kentcdodds/Desktop/my-remix-app/node_modules/start-server-and-test/node_modules/execa/index.js:118:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  shortMessage: 'Command failed with exit code 1: npx cypress run',
  command: 'npx cypress run',
  escapedCommand: '"npx cypress run"',
  exitCode: 1,
  signal: undefined,
  signalDescription: undefined,
  stdout: undefined,
  stderr: undefined,
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
NPM command npm run test:e2e:run -- failed with code 1
```

### LINKED ISSUE #136: Remix.init fails when choosing TypeScript
### Have you experienced this bug with the latest version of the template?

Yes

### Steps to Reproduce

? What type of app do you want to create? **A pre-configured stack ready for production**
? Which Stack do you want? (Learn more about these stacks: 'https://remix.run/stacks') **Indie**
? TypeScript or JavaScript? **TypeScript**
? Do you want me to run `npm install`? **Yes**

💿 Running remix.init script
🚨 Oops, remix.init failed

[index.js:readFileIfNotTypeScript](https://github.com/remix-run/indie-stack/blob/7c23ab436bb2a42fec5a126d14ee7151e6ac9e85/remix.init/index.js#L49) seems to have wrong brackets

Current:
```ts
  (isTypeScript ? Promise.resolve() : fs.readFile(filePath, "utf-8")).then(
    parseFunction
  );
```
Expected
```ts
  (isTypeScript ? Promise.resolve() : fs.readFile(filePath, "utf-8").then(
    parseFunction
  ));
```

### Expected Behavior

remix.init should run successfully

### Actual Behavior

remix.init fails

### LINKED ISSUE #139: create-remix cli run into error when trying to run `npx remix init`
### What version of Remix are you using?

1.6.7

### Steps to Reproduce

Run create-remix cli with options below:

image
****

### Expected Behavior

The cli run `remix.init` script successfully

### Actual Behavior

The `remix.init` script run into error

`🚨 Oops, remix.init failed`
`Cannot read properties of undefined (reading 'length')`

### LINKED ISSUE #140: remix.init failed
### What version of Remix are you using?

1.6.8

### Steps to Reproduce

Hello - First time trying out Remix and the [Getting Started](https://remix.run/docs/en/v1/tutorials/blog) tutorial is immediately failing for me.

macOS Monteray
node version 18.7.0
npm 8.15.0

```sh
npx create-remix --template remix-run/indie-stack jake-start
? TypeScript or JavaScript? TypeScript
? Do you want me to run `npm install`? Yes
```

This is a new machine and I assume you would be aware of a problem like this so may it will end up being a me thing, but everything else seems to be working on my end. 

### Expected Behavior

No CLI errors and "the app should be running," including a 200 at `localhost:3000`.

### Actual Behavior

```sh
🚨 Oops, remix.init failed

Cannot read properties of undefined (reading 'length')
```

More:

```sh
jake:~/repos/remix-getting-started$ npx create-remix --template remix-run/indie-stack jake-start
? TypeScript or JavaScript? TypeScript
? Do you want me to run `npm install`? Yes
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated rollup-plugin-inject@3.0.2: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-inject.
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated

added 1312 packages, and audited 1313 packages in 55s

231 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
💿 Running remix.init script
🚨 Oops, remix.init failed

Cannot read properties of undefined (reading 'length')
```

There ends up being something which looks like an app in the folder, and `npm run dev` initially starts a server, but then loading a page results in:

```
Error: error: Environment variable not found: DATABASE_URL.
  -->  schema.prisma:3
   |
 2 |   provider = "sqlite"
 3 |   url      = env("DATABASE_URL")
```

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
Part of the solution is also a change in the transform, which is implemented in https://github.com/remix-run/remix/pull/3987

---

This will partially resolve #134
Closes #136
Closes #139
Closes #140

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -8,6 +8,10 @@ module.exports = {
     "@remix-run/eslint-config/jest-testing-library",
     "prettier",
   ],
+  env: {
+    "cypress/globals": true,
+  },
```
Comment (MichaelDeBoey) [inline .eslintrc.js:13]: Without this we would either get ESLint errors about `Cypress`, `cy`, ... being undefined _or_ we would need to declare these values ourselves in `globals`

File: cypress/support/commands.ts [hunk 2]
```diff
@@ -93,8 +93,3 @@ function visitAndCheck(url: string, waitTime: number = 1000) {
 Cypress.Commands.add("login", login);
 Cypress.Commands.add("cleanupUser", cleanupUser);
 Cypress.Commands.add("visitAndCheck", visitAndCheck);
-
-/*
-eslint
-  @typescript-eslint/no-namespace: "off",
-*/
```
Comment (MichaelDeBoey) [inline cypress/support/commands.ts:100]: This one's unused, so no need to set it to "off" here

File: remix.init/index.js [hunk 3]
```diff
@@ -10,9 +10,9 @@ const YAML = require("yaml");
 const cleanupCypressFiles = (filesEntries) =>
   filesEntries.flatMap(([filePath, content]) => {
     const newContent = content
-      .replace("npx ts-node", "node")
-      .replace("create-user.ts", "create-user.js")
-      .replace("delete-user.ts", "delete-user.js");
+      .replace(new RegExp("npx ts-node", "g"), "node")
+      .replace(new RegExp("create-user.ts", "g"), "create-user.js")
+      .replace(new RegExp("delete-user.ts", "g"), "delete-user.js");
```
Comment (MichaelDeBoey) [inline remix.init/index.js:15]: We have these values both in code & in comments

File: remix.init/index.js [hunk 4]
```diff
@@ -68,10 +68,7 @@ const updatePackageJson = ({ APP_NAME, isTypeScript, packageJson }) => {
     name: APP_NAME,
     devDependencies: isTypeScript
       ? devDependencies
-      : removeUnusedDependencies(devDependencies, [
-          "ts-node",
-          "vite-tsconfig-paths",
```
Comment (MichaelDeBoey) [inline remix.init/index.js:73]: `vite-tsconfig-paths` is still used now that it supports `jsconfig.json` files

File: .eslintrc.js (hunk 1 above)
Comment (MichaelDeBoey) [inline .eslintrc.js:13]: Without this we would either get ESLint errors about `Cypress`, `cy`, ... being undefined _or_ we would need to declare these values ourselves in `globals`

File: cypress/support/commands.ts (hunk 2 above)
Comment (MichaelDeBoey) [inline cypress/support/commands.ts:100]: This one's unused, so no need to set it to "off" here

File: remix.init/index.js (hunk 3 above)
Comment (MichaelDeBoey) [inline remix.init/index.js:15]: We have these values both in code & in comments

File: remix.init/index.js [hunk 5]
```diff
@@ -46,9 +46,9 @@ const readFileIfNotTypeScript = (
   filePath,
   parseFunction = (result) => result
 ) =>
-  (isTypeScript ? Promise.resolve() : fs.readFile(filePath, "utf-8")).then(
-    parseFunction
-  );
+  isTypeScript
+    ? Promise.resolve()
+    : fs.readFile(filePath, "utf-8").then(parseFunction);
```
Comment (MichaelDeBoey) [inline remix.init/index.js:51]: We don't need to parse the value when we're not reading a file

File: remix.init/index.js [hunk 6]
```diff
@@ -68,12 +68,9 @@ const updatePackageJson = ({ APP_NAME, isTypeScript, packageJson }) => {
     name: APP_NAME,
     devDependencies: isTypeScript
       ? devDependencies
-      : removeUnusedDependencies(devDependencies, [
-          "ts-node",
-          "vite-tsconfig-paths",
```
Comment (MichaelDeBoey) [inline remix.init/index.js:73]: `vite-tsconfig-paths` is still used now that it supports `jsconfig.json` files

File: remix.init/index.js [hunk 7]
```diff
@@ -68,12 +68,9 @@ const updatePackageJson = ({ APP_NAME, isTypeScript, packageJson }) => {
...
-        ]),
+      : removeUnusedDependencies(devDependencies, ["ts-node"]),
     prisma: isTypeScript
-      ? prisma
+      ? { ...prisma, seed: prismaSeed }
```
Comment (MichaelDeBoey) [inline remix.init/index.js:73]: `prisma` is a rest value (without `prismaSeed`), so we need to add the `seed` script again

**Review by kentcdodds — APPROVED** [review]
Good going 👍 Thanks!

**Pull request merged**